CHANGELOG.md

```diff
@@ -4,6 +4,10 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 
 ## Next version
 
+### Changed
+
+- Use array `[String]` for Source and Resource paths https://github.com/tuist/tuist/pull/201 by @dangthaison91
```

Comment:
I've been thinking if we could make this change non-breaking. Could you try doing something like this?

```
extension Array where Element is String : ExpressibleByStringLiteral {
 }
```

The snipped above is pseudocode so I'm not sure if it'll work. The idea is that if someone uses a string instead of an array, it'll initialise an array with only one element, the given string. We should aim for non-breaking changes unless it's strictly necessary.

Reply:
You could maintain source compatibility using something like this:

```swift
struct FileList {
    let sources: [String]
}

extension FileList: ExpressibleByStringLiteral {

    init(stringLiteral value: String) {
        sources = [ value ]
    }
    
}

extension FileList: ExpressibleByArrayLiteral {
    init(arrayLiteral elements: String...) {
        sources = elements
    }
}

let sourcesA: FileList = "abc"
let sourcesB: FileList = [ "a", "b", "c" ]
```

What's your thoughts?

Reply:
Agree with @ollieatkinson's suggestion.

Reply:
I just added `FileList` to maintain backward compatibility as your suggestion, please check @pepibumur @ollieatkinson